pkg/apis/v1alpha1/awsnodetemplate.go

```diff
@@ -104,3 +111,12 @@ type AWSNodeTemplateList struct {
 	metav1.ListMeta `json:"metadata,omitempty"`
 	Items           []AWSNodeTemplate `json:"items"`
 }
+
+// Placement contains the configuration for LaunchTemplates to run instances on dedicated hosts
+// https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-ec2-launchtemplate-launchtemplatedata.html#cfn-ec2-launchtemplate-launchtemplatedata-placement
+type Placement struct {
+	// +optional
+	HostResourceGroupARN *string `json:"hostResourceGroupARN,omitempty"`
+	// +optional
+	GroupID *string `json:"groupdID,omitempty"`
```

Comment:
Do we want this to be ID or name? https://docs.aws.amazon.com/AWSEC2/latest/APIReference/API_CreatePlacementGroup.html. Seems like name is more discoverable.

Reply:
I was basing this off of https://github.com/aws/karpenter/issues/3324#issuecomment-1577360562 I guess we could pull this apart and reimplement with the select model (assuming we can find appropriate aws APIs)